modules/siddhi-extensions/output-mappers/json-output-mapper/src/main/java/org/wso2/siddhi/extension/output/mapper/json/JsonSinkMapper.java

```diff
@@ -0,0 +1,361 @@
+                                + "produce below "
+                                + "output JSON message"
+                                + "{"
+                                + "\"portfolio\":{\n"
+                                + "    \"StockData\":{\n"
+                                + "        \"Symbol\":WSO2,\n"
+                                + "        \"Price\":55.6\n"
+                                + "      }\n"
+                                + "  }\n"
+                                + "}")
+        }
+)
+
+public class JsonSinkMapper extends SinkMapper {
+    private static final Logger log = Logger.getLogger(JsonSinkMapper.class);
+    private static final String EVENT_PARENT_TAG = "event";
+    private static final String ENCLOSING_ELEMENT_IDENTIFIER = "enclosing.element";
+    private static final String DEFAULT_ENCLOSING_ELEMENT = "$";
+    private static final String JSON_VALIDATION_IDENTIFIER = "validate.json";
+    private static final String JSON_EVENT_SEPERATOR = ",";
+    private static final String JSON_KEYVALUE_SEPERATOR = ":";
+    private static final String JSON_ARRAY_START_SYMBOL = "[";
+    private static final String JSON_ARRAY_END_SYMBOL = "]";
+    private static final String JSON_EVENT_START_SYMBOL = "{";
+    private static final String JSON_EVENT_END_SYMBOL = "}";
+    private static final String UNDEFINED = "undefined";
+
+    private String[] attributeNameArray;
+    private String enclosingElement = null;
+    private boolean isJsonValidationEnabled = false;
+
+
+    @Override
+    public String[] getSupportedDynamicOptions() {
+        return new String[0];
+    }
+
+    /**
+     * Initialize the mapper and the mapping configurations.
+     *
+     * @param streamDefinition       The stream definition
+     * @param optionHolder           Option holder containing static and dynamic options
+     * @param payloadTemplateBuilder Unmapped payload for reference
+     */
+    @Override
+    public void init(StreamDefinition streamDefinition, OptionHolder optionHolder,
+                     TemplateBuilder payloadTemplateBuilder, ConfigReader mapperConfigReader) {
+        attributeNameArray = streamDefinition.getAttributeNameArray();
+        enclosingElement = optionHolder.validateAndGetStaticValue(ENCLOSING_ELEMENT_IDENTIFIER, null);
+        isJsonValidationEnabled = Boolean.parseBoolean(optionHolder
+                .validateAndGetStaticValue(JSON_VALIDATION_IDENTIFIER, "false"));
+    }
+
+    @Override
+    public void mapAndSend(Event[] events, OptionHolder optionHolder, TemplateBuilder payloadTemplateBuilder,
+                           SinkListener sinkListener, DynamicOptions dynamicOptions)
+            throws ConnectionUnavailableException {
+        StringBuilder sb = null;
+        if (payloadTemplateBuilder == null) {
+            String jsonString = constructJsonForDefaultMapping(events);
```

Comment:
Since the mappAndProcess only accepts Event and Event[], the return null statements in constructJsonForDefaultMapping(events) will be never reached. So in that case, jsonString can't be null. Will fix it.